### PR TITLE
docs: Template the README file

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
 This is an alternative for automating releases further instead of a bashscript.
It would create an updated readme file based on the content in the data.yaml via
`gomplate -d config=./data.yaml --file README.md.tpl  > README.md`

* [x] Needs CI Integration / Fail if README.md is outdated
* [x] Needs header in README.md that changes need to be made in the template file

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
None
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/assign @rexagod

### CODE REVIEW

**Comment by mrueg** [comment]
@rexagod @dgrisonnet this is now ready for review and should simplify the release process a bit.

File: tools/tools.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	_ "github.com/brancz/gojsontoyaml"
 	_ "github.com/campoy/embedmd"
 	_ "github.com/google/go-jsonnet/cmd/jsonnet"
+	_ "github.com/hairyhenderson/gomplate/v3"
```
Comment (rexagod) [inline tools/tools.go:26]:
```suggestion
	_ "github.com/hairyhenderson/gomplate/v4"
```

I see `v4` is out, can we use that instead?

**Comment by rexagod** [comment]
Whoa! This looks great, and I'm fairly confident it'll be able to handle various templating cases we throw at it with cleanliness, not to mention unblocking us on the release-automation front! Nicely done! 🙌🏼

File: tools/tools.go (hunk 1 above)
Comment (mrueg) [reply tools/tools.go]: v4 still has no official release, just pre-releases. I'd prefer to stay on v3 for now.

**Comment by mrueg** [comment]
@rexagod great to hear, if there are no concerns with this approach, this is ready to go :)

File: README.md [hunk 2]
```diff
@@ -29,6 +29,9 @@ the raw metrics. Note that the metrics exposed on the `/metrics` endpoint
 reflect the current state of the Kubernetes cluster. When Kubernetes objects
 are deleted they are no longer visible on the `/metrics` endpoint.
 
+> [!NOTE]
+> This README is generated from [./README.md.tpl]. Please make changes in the template file.
```
Comment (rexagod) [inline README.md:33]:
```suggestion
> This README is generated from [./README.md.tpl](./README.md.tpl). Please make changes in the template file.
```

Ah, just a nit. We might need to append the path here for it to be hyperlinked.

**Pull request merged**